lib/api/controllers/funnelController.js

```diff
@@ -374,6 +374,17 @@ class FunnelController {
       .then(responseData => {
         modifiedRequest.setResult(responseData, {status: request.status === 102 ? 200 : request.status});
 
+        if (!this.isNativeController(modifiedRequest) && !modifiedRequest.response.raw) {
+          // check if the plugin response can be serialized
+          try {
+            JSON.stringify(modifiedRequest.response.toJSON());
+          }
+          catch (e) {
+            modifiedRequest.setResult(null);
+            throw new Error('Unable to serialize response. Are you trying to return the request?');
```

Comment:
I think it's better to throw a `PluginImplementationError` since the error came from a plugin.  
It's only when returning the original request or any Request object ?

EDIT: in fact the `Error` will be wrapped in a `PluginImplementationError` but still I think it's better to be explicit

Reply:
:+1: